### ISSUE TITLE
Some retrieval datasets don't calculate metadata correctly

### ISSUE THREAD

**Issue by orionw** [embeddings-benchmark/mteb#964]
When running `task.calculate_metadata_metrics()` for retrieval tasks, there are a handful that fail to run (most of the ~130 work though, which is great!)

They are:
- BelebeleRetrieval
- NorQuadRetrieval
- SwednRetrieval
- PublicHealthQA
- CovidRetrieval
- CrossLingualSemanticDiscriminationWMT19
- CrossLingualSemanticDiscriminationWMT21
- DanFEVER
- SweFaqRetrieval
- TV2Nordretrieval
- CodeEditSearchRetrieval
- TwitterHjerneRetrieval
- MLQARetrieval
- SNLRetrieval
- IndicQARetrieval
- XPQARetrieval

Almost all the errors are due to things like:
```
ValueError: BuilderConfig 'corpus' not found. Available: ['default']
```

At some point we should resolve this, either via changes to the `calculate_metadata_metrics` function that use some parameter passed in, or by changing the tasks `__init__` function to define the needed parts.

cc'ing @KennethEnevoldsen as an FYI. The RetrievalStats tab of [this Google Sheet](https://docs.google.com/spreadsheets/d/12Yldu9whVysP0jhnw1S5lmpTi2hOolU2KO5mWgSYk6I/edit?usp=sharing) has the stats for the current tasks that did succeed.  Around 30 will need to be reduced in size, plus some of the above that I do not have stats for potentially.

**Comment by KennethEnevoldsen** [comment]
Thanks for this, Orion - I think most of these are quite small (SNL, Twitterhjerne, TV2Nord, SweFaq, DanFEVER, NorQuad and Swedn)

but the others might be large enough. I will set this to `help wanted` to allow someone to grab it

**Comment by KennethEnevoldsen** [comment]
Seems like some of these are due to missing load_data() functions. I have added these in #953 to ensure that as much of the possible runs for @Muennighoff.

This resolved the issues of the task being unable to load for some of the datasets (SNL, Twitterhjerne, TV2Nord, SweFaq, DanFEVER, NorQuad and Swedn) as well as solve the statistics calculation issue (though not for Norquad). I have to run to another thing so can't look more into it, but this at least solve about half

**Comment by isaac-chung** [comment]
I'll take a look at this.

**Comment by KennethEnevoldsen** [comment]
Seems solved in #965
